functions/src/modules/SQR.functions.ts

```diff
@@ -278,16 +312,39 @@ export const processSubmission = functions.database
   if (allSubmissionsSnapshot.exists()) {
     const allSubmissions = allSubmissionsSnapshot.val();
 
+    const allotments = (await db
+        .ref(`/files/${list}`)
+        .orderByChild("soundQualityReporting/assignee/emailAddress")
+        .equalTo(submission.author.emailAddress).once("value")
+    ).val();
+
+    const currentSet = [];
+
+    Object.entries(allotments).forEach(
+        ([fileName, { languages, soundQualityReporting: { timestampGiven, status } }]: any) => {
+          currentSet.push({
```

Comment:
The idea behind using `forEach` was not to create new array, but to alter items in the same array. Effectively you just need to add `daysPassed` property to the allotment and (probably) modify `timestampGiven` property.